src/sqlfluff/core/parser/parsers.py

```diff
@@ -100,6 +100,47 @@ def match(
         return MatchResult.from_unmatched(segments)
 
 
+class MultiStringParser(StringParser):
+    """An object which matches and returns raw segments on a collection of strings."""
+
+    def __init__(
+        self,
+        templates: Collection[str],
+        raw_class: Type[RawSegment],
+        name: Optional[str] = None,
+        type: Optional[str] = None,
+        optional: bool = False,
+        **segment_kwargs,
+    ):
+        self.templates = {template.upper() for template in templates}
+        super().__init__(
+            template="",
+            raw_class=raw_class,
+            name=name,
+            type=type,
+            optional=optional,
+            **segment_kwargs,
+        )
+        # Delete attribute which is replaced by `self.templates` for this `Parser``
+        del self.template
+
+    def simple(self, parse_context: "ParseContext") -> Optional[List[str]]:
+        """Return simple options for this matcher.
+
+        Because string matchers are not case sensitive we can
+        just return the templates here.
+        """
+        return list(self.templates)
+
+    def _is_first_match(self, segment: BaseSegment):
+        """Does the segment provided match according to the current rules."""
+        # Is the target a match and IS IT CODE.
+        # The latter stops us accidentally matching comments.
+        if segment.is_code and segment.raw.upper() in self.templates:
```

Comment:
Could we use the `raw_upper` property rather than `raw.upper()`? It's a cached property on `BaseSegment`, and since it's cached, could be faster to access.

Reply:
Sounds good - should probably be done `StringParser` too